src/login/test-session-properties.c

```diff
@@ -94,6 +99,34 @@ TEST(set_display) {
         assert_se(isempty(display));
 }
 
+/* Tests org.freedesktop.logind.Session SetTTY */
+TEST(set_tty) {
+        char path[PATH_MAX];
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus* bus = NULL;
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_free_ char *tty = NULL;
+        int fd, r;
+
+        r = ttyname_r(STDIN_FILENO, path, sizeof path);
+        assert_se(r >= 0);
+        fd = open(path, O_RDWR);
```

Comment:
out of principle we require that all open() calls in our codebase come with O_CLOEXEC. (simply so that codeql and such can check for that and complain)

also, maybe also add O_NOCTTY, you deal with a tty here after all